From 3927673571de84d621bdb9e66807a7c6d6c6f2de Mon Sep 17 00:00:00 2001
From: Shunqian Zheng <zhengsq@rock-chips.com>
Date: Tue, 16 Apr 2019 09:01:38 +0800
Subject: [PATCH] media: spi: rk1608: use low spi freq before rk1608
 initialization

rk1608 spi can not run at high speed (>40M) at boot time. It needs
initialization before switching to spi max speed.

Change-Id: I27e17d32271012fd48c7feeb8e4939082439b978
Signed-off-by: Shunqian Zheng <zhengsq@rock-chips.com>
---
 drivers/media/spi/rk1608.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/media/spi/rk1608.c b/drivers/media/spi/rk1608.c
index 6ee7307dc196..36d12b8a7e9d 100644
--- a/drivers/media/spi/rk1608.c
+++ b/drivers/media/spi/rk1608.c
@@ -806,6 +806,7 @@ static int rk1608_power_on(struct rk1608_state *pdata)
 	/* After Reset pull-up, CSn should keep low for 2ms+ */
 	usleep_range(3000, 3500);
 	rk1608_cs_set_value(pdata, 1);
+	rk1608_set_spi_speed(pdata, pdata->min_speed_hz);
 	rk1608_lsb_w32(spi, SPI_ENR, 0);
 	rk1608_lsb_w32(spi, SPI_CTRL0,
 		       OPM_SLAVE_MODE | RSD_SEL_2CYC | DFS_SEL_16BIT);
-- 
2.35.3

